WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -93,6 +93,14 @@ def index
       end
     end
 
+    begin
+      from_date = params[:from_date] && Date.parse(params[:from_date])
+      @competitions = @competitions.where("start_date >= ?", from_date) if from_date
+      to_date = params[:to_date] && Date.parse(params[:to_date])
+      @competitions = @competitions.where("end_date =< ?", to_date) if to_date
+    rescue ArgumentError
```

Comment:
Yikes! I think I know what you're doing here, and we have a [`Date.safe_parse`](https://github.com/thewca/worldcubeassociation.org/blob/62caaf708be4c4d7c29daf2adc7567f6385acaa2/WcaOnRails/config/initializers/monkey_patches.rb#L8) method specifically for this =)